proposals/reward-proposal-v2-reward-distribution.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Epoch Reward Distribution
+---
+
+## Problem
+
+In the `Epoch Reward Calculation V2`, we proposed a new method to handle
+calculating epoch rewards. This is the second part of that proposal - `Reward
+Distribution`.
+
+## Proposed Solutions
+
+Similar to [Partitioned Inflationary Rewards
+Distribution](https://github.com/solana-labs/solana/blob/master/docs/src/proposals/partitioned-inflationary-rewards-distribution.md),
+the rewards are distributed over `M` slots.
+
+To address the concern for the lack of double-entry booking for reward
+distribution, a special system instruction, `DistributeReward`, will be
+introduced to distribute the rewards from the family of `EpochRewardReserve`
+accounts and the `EpochRewardProof` system account.
```

Comment:
What is stored in these `EpochRewardReserve` accounts?

Reply:
The balance and the chunk index. Please see the updated pseudo code.